version: "3"

services:
  app:
    container_name: odoo
    build:
      context: ..
      dockerfile: .devcontainer/Dockerfile
        
    volumes:
      - odoo-web-data:/var/lib/odoo      
      - ../odoo/config:/etc/odoo
      - ../addons:/mnt/extra-addons
      - ..:/workspace:cached      
      
    environment:
    - HOST=db
    - USER=odoo
    - PASSWORD=myodoo
    - DB odoodb
    - OPENERP_SERVER=/etc/odoo/odoo.conf # pytest-odoo config

    # Runs app on the same network as the database container,
    # allows "forwardPorts" in devcontainer.json function
    network_mode: service:db
    depends_on:
      - "db"
  db:
    image: postgres:latest
    container_name: db
    restart: always
    volumes:
      - postgres-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_PASSWORD=myodoo
      - POSTGRES_USER=odoo
      # in local dev-container connection is not localhost, but some internal docker network, see https://stackoverflow.com/a/60262636/4078503
      - POSTGRES_HOST_AUTH_METHOD=trust

  pgadmin:
    container_name: pgadmin4_container
    image: dpage/pgadmin4
    restart: unless-stopped
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@admin.com
      PGADMIN_DEFAULT_PASSWORD: test
      PGADMIN_CONFIG_WTF_CSRF_CHECK_DEFAULT: 'False'
      PGADMIN_CONFIG_WTF_CSRF_ENABLED: 'False'
      PGADMIN_CONFIG_SERVER_MODE: 'False'
      PGADMIN_CONFIG_MASTER_PASSWORD_REQUIRED: 'False'
    depends_on:
      - "db"
    network_mode: service:db
volumes:
  postgres-data:
  odoo-web-data:
